api/src/main/java/jakarta/mail/internet/ParameterList.java

```diff
@@ -429,9 +429,9 @@ private void combineMultisegmentNames(boolean keepConsistentOnFailure)
                                 charset = vv.charset;
                             } else {
                                 if (charset == null) {
-                                    // should never happen
-                                    multisegmentNames.remove(name);
```

Comment:
Rather than throwing, would we just want to update the remove from List to to use `it.remove()` instead. That would fix the CME and maintain intent.  Or is the intent wrong?

Reply:
Yes, I agree. Removing from iterator does not affect the list and would not throw CME. Have changed that and sorry for the late reply.